pkg/southbound/gnmi/client.go

```diff
@@ -31,15 +32,35 @@ type Client interface {
 
 // client gnmi client
 type client struct {
-	client *gclient.Client
+	client  *gclient.Client
+	lock    sync.RWMutex
+	running bool
 }
 
-// Subscribe calls gNMI subscription based on a given query
+// Subscribe calls gNMI subscription bacc
+//sed on a given query
 func (c *client) Subscribe(ctx context.Context, q baseClient.Query) error {
 	err := c.client.Subscribe(ctx, q)
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if !c.running {
```

Comment:
I am trying to understand the usage of this flag at client level. are you trying to achieve one Subscription stream per target?

Reply:
Yes.